servicetalk-dns-discovery-netty/docs/modules/ROOT/pages/index.adoc

```diff
@@ -0,0 +1,71 @@
+// Configure {source-root} values based on how this document is rendered: on GitHub or not
+ifdef::env-github[]
+:source-root:
+endif::[]
+ifndef::env-github[]
+ifndef::source-root[:source-root: https://github.com/apple/servicetalk/blob/{page-origin-refname}]
+endif::[]
+
+= Service Discovery
+
+Another core abstraction in ServiceTalk is its
+link:{source-root}/servicetalk-client-api/src/main/java/io/servicetalk/client/api/ServiceDiscoverer.java[ServiceDiscoverer]
+underpinning the various protocol _Clients_ (eg. HTTP/1.1, HTTP/2, gRPC, etc.).
+
+For scenarios where a client communicates with multiple homogeneous footnote:[belonging to the same service, providing
+same capabilities] server instances, managing addresses of multiple server instances for each _Client_ is complex.
+_ServiceDiscoverer_ makes this easy for users by transparently communicating with mechanisms that know about the actual
+location of remote servers and may provide updates when the network topology is changing.
```

Comment:
"location of remote servers" and "network topology" terms are misleading here as an SD does not really have to know about network topology or physical server locations. As all clients in ST use service discovery I would suggest using the following language:

> A ServiceTalk client leverages _ServiceDiscoverer_ to find server instances that it can use for processing requests. A  _ServiceDiscoverer_  finds server instances associated with a logical name<TODO: Add link to #discover() method>. Additionally, ServiceTalk assumes that the client operates in a dynamic ephemeral environment and the server instances associated with a logical name _may_ change over time. So, a _ServiceDiscoverer_ returns a `Publisher` of server instances and their current relationship to the logical name i.e. whether it is added or removed from the logical name. Using this information a _Client_ can maintain an active collection of server instances that it can use.

Reply:
> <TODO: Add link to #discover() method>

It's hard to add a link to the line number of the source code file because the link will be broken when we update the source code. I slightly reworded your suggestion to include `ServiceDiscovererEvents`.